feat(viewport-panes): add viewport-pane component

jefflembeck opened the pull request:
This gives us the ability to add a class to the top pane of the layout
and have it scale based on the size of the viewport's height.

[Finishes: #113681439]

BREAKING CHANGE: Adds variables that will affect the header as well

jefflembeck committed:
feat(viewport-panes): add viewport-pane component

This gives us the ability to add a class to the top pane of the layout
and have it scale based on the size of the viewport's height.

[Finishes: #113681439]

BREAKING CHANGE: Adds variables that will affect the header as well

stubbornella commented on src/pivotal-ui/components/pui-variables.scss:
```diff
@@ -1326,4 +1326,7 @@ $header-subtle-text-color: rgba(255,255,255,0.8);
\ No newline at end of file
+$greigh3: rgba(black, 0.6);
+
+$header-height-max-sm: 7.478125em;
```
hehe. :)

aredridel replied:
Oh god these magic numbers!

jefflembeck commented:
There are some bizarre numbers in here. They come from the great land of measurement of how tall the header is.

This is unfortunate, as there are small bugs in the header which will undoubtedly affect its size.

We can update things in order to fix this, and likely will have to, now that I've set height restrictions on the header itself, but coming up with an ideal solution gets in the way of having a solution that works right now, and this does that.

stubbornella commented on src/pivotal-ui/components/viewport-panes/viewport-panes.scss:
```diff
@@ -0,0 +1,63 @@
+*/
+
+.viewport-pane {
+  height: calc(100vh - $header-height-max-sm);
```
clever monkey! :monkey: love it

stubbornella commented on src/pivotal-ui/components/viewport-panes/viewport-panes.scss:
```diff
@@ -0,0 +1,63 @@
+
+@media (min-width: $screen-sm-min) {
+  .viewport-pane {
+    height: calc(80vh - $header-height-max-lg);
```
I think the math may be slightly off here. It needs to leave room for both the header and the enterprise pane below it (the word "enterprise" in the next pane needs to show).

Actually wait, this is perfect for the one on the On-Site landing page. We just need an additional one for the homepage.

stubbornella replied:
also, I think this should be min-height. In the unfortunate case where we have too much content for on screen height, we don't want things to get cropped.

aredridel replied:
Ooh good call

stubbornella commented:
This is rad and will work perfectly (I think, though I didn't pull it and test it) on the on-site landing page, we just need to make one that isn't quite so tall and leaves room to see the enterprise secondary hero on the homepage.

stubbornella commented:
Can you upload a screen grab of the result that includes your browser chrome?

jefflembeck committed:
fix(viewport-panes): height -> min-height

Make these min-height so as the content won't get squeezed in, in the case of us having more content than would fit on a page

jefflembeck committed:
fix(viewport-panes): enterprise work

80% fix for making sure enterprise shows up in larger height viewports

This will work for the time being, we can come up with a better fix
later

jefflembeck committed:
fix(viewport-panes): interpolation

calc requires variables be interpolated in sass

jefflembeck committed:
fix(viewport-panes): Smaller wide screens

This really shouldn't be width based. It should be height based, and
while that height is a bit arbitrary, it should definitely affect
screens that are on a 13" laptop

jefflembeck commented:
![bring-the-pane](https://cloud.githubusercontent.com/assets/109699/13716721/c06575ae-e791-11e5-8d14-b543774162a2.gif)

aredridel commented:
Can confirm: pane both paney and viewporty: 
<img width="1322" alt="screen shot 2016-03-11 at 16 41 47" src="https://cloud.githubusercontent.com/assets/2876/13716760/226b3946-e7ab-11e5-8eb3-ce38a5c81be2.png">
